File: src/main/java/com/natia/secretmod/features/dungeons/BonzoSpiritHook.java
package com.natia.secretmod.features.dungeons;

import com.google.common.base.Stopwatch;
import com.natia.secretmod.SecretUtils;
import com.natia.secretmod.config.SkyTweaksConfig;
import com.natia.secretmod.core.SkyblockTimer;
import com.natia.secretmod.features.TimersHook;
import com.natia.secretmod.utils.ItemUtils;
import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.util.EnumChatFormatting;
import net.minecraft.util.StringUtils;
import net.minecraftforge.client.event.ClientChatReceivedEvent;

public class BonzoSpiritHook {
    Minecraft mc = Minecraft.getMinecraft();
    SkyblockTimer skyblockTimer = null;
    Stopwatch sbStopwatch = Stopwatch.createUnstarted();

    public void chat(ClientChatReceivedEvent event) {
        if (!SkyTweaksConfig.maskTimers) return;

        /* I dont know exact message lmao */
        String message = StringUtils.stripControlCodes(event.message.getUnformattedText());
        if (!message.contains(": ") && message.endsWith("saved your life!")) {
            /* yoinked from Danker's Skyblock Mod.
            * https://github.com/bowser0000/SkyblockMod/
             */
            EntityPlayerSP player = mc.thePlayer;
            ItemStack bonzoMask = player.getCurrentArmor(3);
            if (bonzoMask != null && bonzoMask.getItem() == Items.skull) {
                for (String line : ItemUtils.getLore(bonzoMask)) {
                    String stripped = StringUtils.stripControlCodes(line);
                    if (stripped.startsWith("Cooldown: ")) {
                        if (sbStopwatch.isRunning()) sbStopwatch.reset();
                        sbStopwatch.start();

                        cooldownSeconds = Integer.parseInt(stripped.replace("Cooldown: ", "").replace("s", ""));
                        skyblockTimer = new SkyblockTimer(sbStopwatch,
                                EnumChatFormatting.GOLD + "Bonzo Mask Cooldown", cooldownSeconds, false, SecretUtils.generateEmptyRunnable());
                        TimersHook.addTimer(skyblockTimer);
                        break;
                    }
                }
            }
        }
    }
    int cooldownSeconds = 999999999;

    private static BonzoSpiritHook INSTANCE = new BonzoSpiritHook();
    public static BonzoSpiritHook getInstance() {
        return INSTANCE;
    }

}

File: src/main/kotlin/natia/skytweaks/features/dungeons/BonzoSpiritHook.kt
package natia.skytweaks.features.dungeons

import com.google.common.base.Stopwatch
import mixin.natia.skytweaks.SkyTweaksConfig
import natia.skytweaks.core.SkyblockTimer
import natia.skytweaks.features.TimersHook
import natia.skytweaks.utils.ItemUtils
import natia.skytweaks.SecretUtils
import net.minecraft.client.Minecraft
import net.minecraft.init.Items
import net.minecraft.util.EnumChatFormatting
import net.minecraft.util.StringUtils
import net.minecraftforge.client.event.ClientChatReceivedEvent

class BonzoSpiritHook {
    internal var mc = Minecraft.getMinecraft()
    internal var skyblockTimer: SkyblockTimer? = null
    internal var sbStopwatch = Stopwatch.createUnstarted()
    internal var cooldownSeconds = 999999999

    fun chat(event: ClientChatReceivedEvent) {
        if (!SkyTweaksConfig.maskTimers) return

        /* I dont know exact message lmao */
        val message = StringUtils.stripControlCodes(event.message.unformattedText)
        if (!message.contains(": ") && message.endsWith("saved your life!")) {
            /* yoinked from Danker's Skyblock Mod.
            * https://github.com/bowser0000/SkyblockMod/
             */
            val player = mc.thePlayer
            val bonzoMask = player.getCurrentArmor(3)
            if (bonzoMask != null && bonzoMask.item === Items.skull) {
                for (line in ItemUtils.getLore(bonzoMask)) {
                    val stripped = StringUtils.stripControlCodes(line)
                    if (stripped.startsWith("Cooldown: ")) {
                        if (sbStopwatch.isRunning) sbStopwatch.reset()
                        sbStopwatch.start()

                        cooldownSeconds = Integer.parseInt(stripped.replace("Cooldown: ", "").replace("s", ""))
                        skyblockTimer = SkyblockTimer(sbStopwatch,
                                EnumChatFormatting.GOLD.toString() + "Bonzo Mask Cooldown", cooldownSeconds, false, SecretUtils.generateEmptyUnit())
                        TimersHook.addTimer(skyblockTimer!!)
                        break
                    }
                }
            }
        }
    }

    companion object {

        val instance = BonzoSpiritHook()
    }

}
